AIAgent/run_training.py

```diff
@@ -70,72 +69,104 @@ class TrialSettings:
+
+        def load_maps_info():
+            maps: list[GameMap2SVM] = list()
+            for platform in validation_config.validation.platforms_config:
+                svms_info = platform.svms_info
+                for svm_info in svms_info:
+                    for dataset_config in platform.dataset_configs:
+                        dataset_base_path = dataset_config.dataset_base_path
+                        dataset_description = dataset_config.dataset_description
+                        with open(dataset_description, "r") as maps_json:
+                            single_json_maps: list[GameMap] = GameMap.schema().load(
+                                json.loads(maps_json.read()), many=True
+                            )
+                        single_json_maps_with_svms: list[GameMap2SVM] = list()
+                        for _map in single_json_maps:
+                            fullName = os.path.join(
+                                dataset_base_path, _map.AssemblyFullName
+                            )
+                            _map.AssemblyFullName = fullName
+                            single_json_maps_with_svms.append(
+                                GameMap2SVM(_map, svm_info)
+                            )
+                        maps.extend(single_json_maps_with_svms)
+            return maps
+
+        maps = load_maps_info()
+        with open(CURRENT_TABLE_PATH, "w") as statistics_file:
+            statistics_writer = csv.DictWriter(
+                statistics_file,
+                sorted([game_map2svm.GameMap.MapName for game_map2svm in maps]),
+            )
+            statistics_writer.writeheader()
 
-    maps: list[GameMap2SVM] = list()
-    for platform in platforms_config:
-        svms_info = platform.svms_info
-        for svm_info in svms_info:
-            for dataset_config in platform.dataset_configs:
-                dataset_base_path = dataset_config.dataset_base_path
-                dataset_description = dataset_config.dataset_description
-                with open(dataset_description, "r") as maps_json:
-                    single_json_maps: list[GameMap] = GameMap.schema().load(
-                        json.loads(maps_json.read()), many=True
-                    )
-                single_json_maps_with_svms: list[GameMap2SVM] = list()
-                for _map in single_json_maps:
-                    fullName = os.path.join(dataset_base_path, _map.AssemblyFullName)
-                    _map.AssemblyFullName = fullName
-                    single_json_maps_with_svms.append(GameMap2SVM(_map, svm_info))
-                maps.extend(single_json_maps_with_svms)
+        def validate(model, dataset: TrainingDataset):
+            result, metrics, maps_to_remove = validate_coverage(
+                model, dataset, maps, validation_config.validation
+            )
+            if maps_to_remove and validation_config.validation.fail_immediately:
```

Comment:
Look at [this code](https://github.com/PySymGym/PySymGym/pull/120/files#diff-3f4e39c863773963ee6442034b7b73bcac4f3003ea83f881cdafaf824ecf3be7R85). `fail_immediate`  must be independent from other flags. CI must fails on any problems with maps in any cases.